### PR TITLE
Add a few more model params

### PR DESCRIPTION
**Pull request by ankrgyl**

### CODE REVIEW

File: core/js/typespecs/prompt.ts [hunk 1]
```diff
@@ -80,6 +80,15 @@ const openAIModelParamsSchema = z
           .strip(),
       ])
       .optional(),
+    function_call: z.union([
+      z.literal("auto"),
+      z.literal("none"),
+      z.strictObject({
```
Comment (manugoyal) [inline core/js/typespecs/prompt.ts:86]: Should we add `.strip()` here if we are unsure whether there might be more properties than `name`?

**Review by manugoyal — APPROVED** [review]
lgtm. Out of curiosity, was the absence of these params causing zod errors? Because I would hope that the `.strip()` annotation will make zod strip out any extraneous fields rather than error.

File: core/js/typespecs/prompt.ts (hunk 1 above)
Comment (ankrgyl) [reply core/js/typespecs/prompt.ts:86]: I think it's better to keep it strict, especially since this is a deprecated parameter.

**Comment by ankrgyl** [comment]
> Out of curiosity, was the absence of these params causing zod errors? 

Nope, it was just preventing some param translation stuff from typechecking in the proxy.

**Pull request merged**